lib/galaxy/tool_util/deps/mulled/mulled_build.py

```diff
@@ -284,9 +286,18 @@ def mull_targets(
         involucro_args.extend(["-set", f"USER_ID={os.getuid()}:{os.getgid()}"])
     if test:
         involucro_args.extend(["-set", f"TEST={test}"])
-    if conda_version is not None:
-        verbose = "--verbose" if verbose else "--quiet"
-        involucro_args.extend(["-set", f"PREINSTALL=conda install {verbose} --yes conda={conda_version}"])
+
+    verbose = "--verbose" if verbose else "--quiet"
+    conda_bin = "conda"
+    if use_mamba:
+        constraint = "" if mamba_version is None else f"={mamba_version}"
+        involucro_args.extend(["-set", f"PREINSTALL=conda install {verbose} --yes mamba{constraint}"])
+        conda_bin = "mamba"
+    else:
+        if conda_version is not None:
+            involucro_args.extend(["-set", f"PREINSTALL=conda install {verbose} --yes conda={conda_version}"])
+    involucro_args.extend(["-set", "CONDA_BIN=%s" % conda_bin])
```

Comment:
```suggestion
    if use_mamba:
        if mamba_version is None:
            mamba_version = ""
        conda_bin = "mamba"
    if conda_version is not None or mamba_version is not None:
        mamba_test = "true"
        specs = []
        if conda_version is not None:
            specs.append(f"conda={conda_version}")
        if mamba_version is not None:
            specs.append(f"mamba={mamba_version}")
            if mamba_version == "" and not specs:
                # If nothing but mamba without a specific version is requested,
                # then only run conda install if mamba is not already installed.
                mamba_test = f"[ '[]' = \"$( conda list --json --full-name mamba )\" ]"
        conda_install = f"""conda install {verbose} --yes {" ".join(f"'{spec}'" for spec in specs)}"""
        involucro_args.extend(["-set", f"PREINSTALL=if {mamba_test} ; then {conda_install} ; fi"]}
    involucro_args.extend(["-set", "CONDA_BIN=%s" % conda_bin])
```
Changes:
- Decouple `use_mamba` and the version requirements (e.g., one could use `mamba`, but still require a specific `conda` version). Also, if both `conda_version` and `mamba_version` are given, still have only one `conda install` invocation.
- Even if `mamba_version` is `None`, we can set append an empty constraint `=` since `conda install` handles `mamba=` and `mamba` equivalently.
- Skip `conda install mamba` if `mamba` is already installed. This is, e.g., the case in the `quay.io/bioconda/create-env` container image we use in `bioconda-utils`.